pkg/utils/utils_test.go

```diff
@@ -1062,4 +1063,206 @@ var _ = Describe("Utils", Label("utils"), func() {
 			Expect(utils.IsUkiWithFs(fs)).To(BeFalse())
 		})
 	})
+	Describe("AddBootAssessment", func() {
+		BeforeEach(func() {
+			Expect(fsutils.MkdirAll(fs, "/efi/loader/entries", os.ModePerm)).ToNot(HaveOccurred())
+		})
+		It("adds the boot assessment to a file", func() {
+			err := fs.WriteFile("/efi/loader/entries/test.conf", []byte(""), os.ModePerm)
+			Expect(err).ToNot(HaveOccurred())
+			err = utils.AddBootAssessment(fs, "/efi/loader/entries", logger)
+			Expect(err).ToNot(HaveOccurred())
+			Expect("/efi/loader/entries/test.conf").ToNot(matchers.BeAnExistingFileFs(fs))
+			// Should match with the +3
+			Expect("/efi/loader/entries/test+3.conf").To(matchers.BeAnExistingFileFs(fs))
+		})
+		It("adds the boot assessment to several files", func() {
+			err := fs.WriteFile("/efi/loader/entries/test1.conf", []byte(""), os.ModePerm)
+			Expect(err).ToNot(HaveOccurred())
+			err = fs.WriteFile("/efi/loader/entries/test2.conf", []byte(""), os.ModePerm)
+			Expect(err).ToNot(HaveOccurred())
+			err = fs.WriteFile("/efi/loader/entries/test3.conf", []byte(""), os.ModePerm)
+			Expect(err).ToNot(HaveOccurred())
+			err = utils.AddBootAssessment(fs, "/efi/loader/entries", logger)
+			Expect(err).ToNot(HaveOccurred())
+			Expect("/efi/loader/entries/test1.conf").ToNot(matchers.BeAnExistingFileFs(fs))
+			Expect("/efi/loader/entries/test2.conf").ToNot(matchers.BeAnExistingFileFs(fs))
+			Expect("/efi/loader/entries/test3.conf").ToNot(matchers.BeAnExistingFileFs(fs))
+			// Should match with the +3
+			Expect("/efi/loader/entries/test1+3.conf").To(matchers.BeAnExistingFileFs(fs))
+			Expect("/efi/loader/entries/test2+3.conf").To(matchers.BeAnExistingFileFs(fs))
+			Expect("/efi/loader/entries/test3+3.conf").To(matchers.BeAnExistingFileFs(fs))
+		})
+		It("leaves assessment in place for existing files", func() {
+			err := fs.WriteFile("/efi/loader/entries/test1.conf", []byte(""), os.ModePerm)
+			Expect(err).ToNot(HaveOccurred())
+			err = fs.WriteFile("/efi/loader/entries/test2+3.conf", []byte(""), os.ModePerm)
+			Expect(err).ToNot(HaveOccurred())
+			err = fs.WriteFile("/efi/loader/entries/test3+1-2.conf", []byte(""), os.ModePerm)
+			Expect(err).ToNot(HaveOccurred())
+			err = utils.AddBootAssessment(fs, "/efi/loader/entries", logger)
+			Expect(err).ToNot(HaveOccurred())
+			Expect("/efi/loader/entries/test1.conf").ToNot(matchers.BeAnExistingFileFs(fs))
+			Expect("/efi/loader/entries/test3+3.conf").ToNot(matchers.BeAnExistingFileFs(fs))
+			// Should match with the +3 and the existing ones left in place
+			Expect("/efi/loader/entries/test1+3.conf").To(matchers.BeAnExistingFileFs(fs))
+			Expect("/efi/loader/entries/test2+3.conf").To(matchers.BeAnExistingFileFs(fs))
+			Expect("/efi/loader/entries/test3+1-2.conf").To(matchers.BeAnExistingFileFs(fs))
+		})
+		It("fails to write the boot assessment in non existing dir", func() {
+			err := utils.AddBootAssessment(fs, "/fake", logger)
+			Expect(err).To(HaveOccurred())
```

Comment:
Better use `MatchError` to match the specific error. Just in case something else fails in the future instead of the expected error.